Cannot build with Version 7.0.0

Here's log : 

> `e: C:\flutter\.pub-cache\hosted\pub.flutter-io.cn\stripe_android-7.0.0\android\src\main\kotlin\com\reactnativestripesdk\Mappers.kt: (453, 3): 'when' expression must be exhaustive, add necessary 'UpiAwaitNotification' branch or 'else' branch instead
e: C:\flutter\.pub-cache\hosted\pub.flutter-io.cn\stripe_android-7.0.0\android\src\main\kotlin\com\reactnativestripesdk\utils\Mappers.kt: (458, 3): 'when' expression must be exhaustive, add necessary 'UpiAwaitNotification' branch or 'else' branch instead

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':stripe_android:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 47s
Exception: Gradle task assembleDebug failed with exit code 1
`

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlin_version = '1.6.21'
+    ext.kotlin_version = '1.7.10'
     repositories {
         google()
         mavenCentral()

--- a/example/pubspec.yaml
+++ b/example/pubspec.yaml
@@ -10,7 +10,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  flutter_stripe: ^6.0.0
+  flutter_stripe: ^7.0.0
   flutter_stripe_web: ^2.0.0
   stripe_checkout: ^1.0.0
   pay: ^1.0.10

--- a/packages/stripe/pubspec.yaml
+++ b/packages/stripe/pubspec.yaml
@@ -21,7 +21,7 @@ flutter:
 dependencies:
   flutter:
     sdk: flutter
-  stripe_android: ^7.0.0
+  stripe_android: ^7.0.1
   stripe_ios: ^7.0.0
   stripe_platform_interface: ^7.0.0
 dev_dependencies:

--- a/packages/stripe_android/CHANGELOG.md
+++ b/packages/stripe_android/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 7.0.1
+
+ - Fix #1022 make library compatible with Kotlin 1.7
+
 ## 7.0.0
 **Breaking Changes**
 - This library now supports iOS 13 and up, due to stripe-ios increasing the deployment target

--- a/packages/stripe_android/android/src/main/kotlin/com/reactnativestripesdk/Mappers.kt
+++ b/packages/stripe_android/android/src/main/kotlin/com/reactnativestripesdk/Mappers.kt
@@ -482,7 +482,7 @@ internal fun mapNextAction(type: NextActionType?, data: NextActionData?): Writab
     NextActionType.AlipayRedirect -> { // TODO: Can't access, private
       return null
     }
-    NextActionType.BlikAuthorize, NextActionType.UseStripeSdk, null -> {
+    NextActionType.BlikAuthorize, NextActionType.UseStripeSdk, NextActionType.UpiAwaitNotification,  null -> {
       return null
     }
   }

--- a/packages/stripe_android/android/src/main/kotlin/com/reactnativestripesdk/utils/Mappers.kt
+++ b/packages/stripe_android/android/src/main/kotlin/com/reactnativestripesdk/utils/Mappers.kt
@@ -487,7 +487,7 @@ internal fun mapNextAction(type: NextActionType?, data: NextActionData?): Writab
     NextActionType.AlipayRedirect -> { // TODO: Can't access, private
       return null
     }
-    NextActionType.BlikAuthorize, NextActionType.UseStripeSdk, null -> {
+    NextActionType.BlikAuthorize, NextActionType.UseStripeSdk, NextActionType.UpiAwaitNotification,  null -> {
       return null
     }
   }

--- a/packages/stripe_android/pubspec.yaml
+++ b/packages/stripe_android/pubspec.yaml
@@ -1,6 +1,6 @@
 name: stripe_android
 description: Stripe platform implementation for Android
-version: 7.0.0
+version: 7.0.1
 repository: https://github.com/flutter-stripe/flutter_stripe
 homepage: https://pub.dev/packages/flutter_stripe
 

